### PR TITLE
test: fix jest usage and add retry

### PR DESCRIPTION
**Pull request by ylemkimon**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [x] This is a **test update**
- [ ] This is a **docs update**
- [x] This is a **metadata update**

### For Bugs and Features; did you add new tests?
N/A

### Motivation / Use-Case
- Update `jest` to 26.6.3
- Use `jest-circus` runner
- Fix jest usages
- Retry test three times for flaky tests, e.g., `puppeteer`

### Breaking Changes
N/A

### CODE REVIEW

File: test/e2e/ClientOptions.test.js [hunk 1]
```diff
@@ -400,7 +400,7 @@ describe('Client console.log', () => {
       })`;
       options = { ...mode, ...options };
       const testOptions = Object.assign({}, baseOptions, options);
-      await it(title, async (done) => {
+      await it(title, (done) => {
```
Comment (ylemkimon) [inline test/e2e/ClientOptions.test.js:403]: > Test functions cannot both take a 'done' callback and return something. Either use a 'done' callback, or return a promise.

File: test/options.test.js [hunk 2]
```diff
@@ -44,9 +44,9 @@ describe('options', () => {
 
     afterAll((done) => {
       if (server) {
-        server.close(() => {
-          done();
-        });
+        server.close(done);
+      } else {
+        done();
```
Comment (ylemkimon) [inline test/options.test.js:49]: `done` was not called if the `server` was `null`, causing `Exceeded timeout of 20000 ms for a hook.`.

File: test/server/https-option.test.js [hunk 3]
```diff
@@ -42,9 +42,7 @@ describe('https option', () => {
       req.get('/').expect(200, /Heyo/, done);
     });
 
-    afterAll(() => {
-      testServer.close();
-    });
```
Comment (ylemkimon) [inline test/server/https-option.test.js:47]: `testServer.close` requires a callback, causing `TypeError: done is not a function`.

File: test/server/utils/runBonjour.test.js [hunk 4]
```diff
@@ -9,34 +9,29 @@ jest.mock('os', () => {
 });
 
 describe('runBonjour', () => {
-  let mock;
-  let publish = jest.fn();
-  let unpublishAll = jest.fn();
```
Comment (ylemkimon) [inline test/server/utils/runBonjour.test.js:14]: Variables referenced outside of `jest.mock()` should be prefixed with `mock`.

**Comment by evilebottnawi** [comment]
Big thanks for helping

**Review by evilebottnawi — APPROVED** [review]

**Pull request merged**